File: review-engine/examples/rule_self_tests/cases/java/shared_security/compliant.java
class SecurityReview {
  private static final String apiKey = loadSecret("service_api_key");

  void loadUser(Connection conn, String userId) throws Exception {
    new ProcessBuilder("id", userId).start();
    String query = "SELECT * FROM users WHERE id = ?";
    PreparedStatement statement = conn.prepareStatement(query);
    statement.setString(1, userId);
    statement.executeQuery();
  }
}

File: review-engine/examples/rule_self_tests/cases/go/shared_security/compliant.go
package security

import (
	"os"
	"os/exec"
)

var api_key = os.Getenv("SERVICE_API_KEY")

func loadUser(db DB, userID string) error {
	_ = exec.Command("id", userID).Run()
	query := "SELECT * FROM users WHERE id = ?"
	return db.Query(query, userID)
}
